app/portainer/settings/authentication/internal-auth/InternalAuth.tsx

```diff
@@ -0,0 +1,44 @@
+import { FormSectionTitle } from '@/portainer/components/form-components/FormSectionTitle';
+import { react2angular } from '@/react-tools/react2angular';
+
+import { SaveAuthSettingsButton } from '../components/SaveAuthSettingsButton';
+
+import { PasswordLengthSlider } from './PasswordLengthSlider';
+
+export interface Props {
+  onSaveSettings(): void;
+  saveButtonState: boolean;
+}
+
+export function InternalAuth({ onSaveSettings, saveButtonState }: Props) {
+  return (
```

Comment:
where is the state of the form? how do you save it?

Reply:
passed from parent component and is used to determine whether to show the save button with a loading icon
https://github.com/portainer/portainer/blob/12cddbd89682aae8b974cca83447965b53dcb5ca/app/portainer/views/settings/authentication/settingsAuthentication.html#L34